### ISSUE TITLE
page:transition:start hook not being called?

### ISSUE THREAD

**Issue by minkoonim** [nuxt/nuxt#23377]
### Environment

- Operating System: Linux
- Node Version:     v18.17.1
- Nuxt Version:     3.7.3
- CLI Version:      3.9.0
- Nitro Version:    2.6.4-28243740.058bacf
- Package Manager:  bun@1.0.3
- Builder:          -
- User Config:      app, site, typescript, devtools, modules, nitro, devServer, runtimeConfig, build, imports, css, pinia, piniaPersistedstate, i18n, device
- Runtime Modules:  @pinia/nuxt@0.4.11, @pinia-plugin-persistedstate/nuxt@1.1.2, @nuxtjs/i18n@8.0.0-rc.4, @nuxtseo/module@2.0.0-beta.31, @nuxtjs/device@3.1.1, @vueuse/nuxt@10.4.1, @unocss/nuxt@0.56.1, @formkit/nuxt@1.1.0, nuxt-icon@0.5.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-vxr1cj?file=app.vue,plugins%2Fhook.ts

### Describe the bug

page:transition:start hook not working
page:transition:end working

```
export default defineNuxtPlugin(({ vueApp, hook, hooks }) => {
  hook('page:transition:start', () => {
    alert('start');
  });

  hook('page:transition:finish', () => {
    alert('end');
  });
});
```

### Additional context

I think it was working days before

**Comment by minkoonim** [comment]
I see RuntimeNuxtHooks in nuxt.d.ts has 'page:transition:start' but not in docs,
is it gone now?

**Comment by danielroe** [comment]
I don't believe this has ever been included in Nuxt 3, so we should either implement it or remove it from the docs.

For layouts (and pages for that matter), you can pass `onBeforeEnter `/`onAfterLeave` as options, e.g. `<NuxtLayout :transition="{ onBeforeEnter() {} }">`.

**Comment by minkoonim** [comment]
> I don't believe this has ever been included in Nuxt 3, so we should either implement it or remove it from the docs.
> 
> For layouts (and pages for that matter), you can pass `onBeforeEnter `/`onAfterLeave` as options, e.g. `<NuxtLayout :transition="{ onBeforeEnter() {} }">`.

![image](https://github.com/nuxt/nuxt/assets/38183949/65851e83-a8f0-4b7c-8654-eaf168f70a5b)

yeop, types should be updated.
why not add to docs <NuxtLayout> and <NuxtPage> transition props usage.
I think below has more explicit control over page transition 
than setting inside app.pageTransition of nuxt.config

NuxtPage has transition props, NuxtLayout doesn't have one.

```
<NuxtPage
    :transition="{
        name: 'page',
        mode: 'out-in',
        onAfterLeave(){},
        onAfterEnter() {},
        ...
      }"
  />
```

```
<Transition
  v-bind="{
    mode: 'out-in',
    name: 'layout',
    onAfterLeave,
    ...
  }"
>
  <div id="layout" :key="currentRoute.meta.layout || 'default'">
    <NuxtLayout />
  </div>
</Transition>
```

**Comment by minkoonim** [comment]
this doesn't work
```
<NuxtLayout
  :transition="{
    mode: 'out-in',
    name: 'layout',
    onAfterEnter() {
      alert('after enter')
    },
  }"
/>
```

**Comment by Mini-ghost** [comment]
`page:transition:start` has already been removed from the type definitions in #31040, so I will go ahead and close this issue.
